src/global/tracking/TrackPropagation_factory.cc

```diff
@@ -0,0 +1,183 @@
+	const double BEMC_halfz = (std::max(m_geoSvc->detector()->constant<double>("EcalBarrelBackward_zmax"), m_geoSvc->detector()->constant<double>("EcalBarrelForward_zmax")) / dd4hep::mm) * extend;
+	auto BEMC_Trf         = transform * Acts::Translation3(Acts::Vector3(0, 0, 0));
+	m_BEMC_prop_surface = Acts::Surface::makeShared<Acts::CylinderSurface>(BEMC_Trf, BEMC_R, BEMC_halfz);
+	m_target_surface_list.push_back(m_BEMC_prop_surface);
+	m_target_surface_ID.push_back(m_geoSvc->detector()->constant<int32_t>("ECalBarrel_ID"));
+
+	// Create propagation surface for LFHCAL
+	std::shared_ptr<Acts::DiscSurface> m_LFHCAL_prop_surface;
+	const double LFHCAL_Z    = (m_geoSvc->detector()->constant<double>("HcalEndcapP_zmin") / dd4hep::mm) + HCAL_avgClusterDepth;
+	const double LFHCAL_MinR = 0.0;
+	const double LFHCAL_MaxR = (m_geoSvc->detector()->constant<double>("HcalEndcapP_rmax") / dd4hep::mm) * extend;
+	auto LFHCAL_Bounds       = std::make_shared<Acts::RadialBounds>(LFHCAL_MinR, LFHCAL_MaxR);
+	auto LFHCAL_Trf          = transform * Acts::Translation3(Acts::Vector3(0, 0, LFHCAL_Z));
+	m_LFHCAL_prop_surface    = Acts::Surface::makeShared<Acts::DiscSurface>(LFHCAL_Trf, LFHCAL_Bounds);
+	m_target_surface_list.push_back(m_LFHCAL_prop_surface);
+	m_target_surface_ID.push_back(m_geoSvc->detector()->constant<int32_t>("HCalEndcapP_ID"));
+
+	// Create propagation surface for OHCAL
+	std::shared_ptr<Acts::CylinderSurface> m_OHCAL_prop_surface;
+	const double OHCAL_R     = (m_geoSvc->detector()->constant<double>("HcalBarrel_rmin") / dd4hep::mm) + HCAL_avgClusterDepth;
+	const double OHCAL_halfz = (std::max(m_geoSvc->detector()->constant<double>("HcalBarrelBackward_zmax"), m_geoSvc->detector()->constant<double>("HcalBarrelForward_zmax")) / dd4hep::mm) * extend;
+	auto OHCAL_Trf           = transform * Acts::Translation3(Acts::Vector3(0, 0, 0));
+	m_OHCAL_prop_surface = Acts::Surface::makeShared<Acts::CylinderSurface>(OHCAL_Trf, OHCAL_R, OHCAL_halfz);
+	m_target_surface_list.push_back(m_OHCAL_prop_surface);
+	m_target_surface_ID.push_back(m_geoSvc->detector()->constant<int32_t>("HCalBarrel_ID"));
+
+	// Create propagation surface for EHCAL
+	std::shared_ptr<Acts::DiscSurface> m_EHCAL_prop_surface;
+	const double EHCAL_Z    = -(m_geoSvc->detector()->constant<double>("HcalEndcapN_zmin") / dd4hep::mm) - HCAL_avgClusterDepth;
+	const double EHCAL_MinR = 0.0;
+	const double EHCAL_MaxR = (m_geoSvc->detector()->constant<double>("HcalEndcapN_rmax") / dd4hep::mm) * extend;
+	auto EHCAL_Bounds       = std::make_shared<Acts::RadialBounds>(EHCAL_MinR, EHCAL_MaxR);
+	auto EHCAL_Trf          = transform * Acts::Translation3(Acts::Vector3(0, 0, EHCAL_Z));
+	m_EHCAL_prop_surface  = Acts::Surface::makeShared<Acts::DiscSurface>(EHCAL_Trf, EHCAL_Bounds);
+	m_target_surface_list.push_back(m_EHCAL_prop_surface);
+	m_target_surface_ID.push_back(m_geoSvc->detector()->constant<int32_t>("HCalEndcapN_ID"));
+
+	/*
+	std::cout << std::endl;
```

Comment:
Please don't add commented code. Either convert it to logger debug messages or remove entirely.

Reply:
Sure, I think this is useful to have in so I will make it a debug message.